target/linux/ath79/image/generic.mk

```diff
@@ -363,6 +382,26 @@ define Device/avm_fritzdvbc
 endef
 TARGET_DEVICES += avm_fritzdvbc
 
+define Device/belkin_f9j1108-v2
+  SOC := qca9558
+  DEVICE_VENDOR := Belkin
+  DEVICE_MODEL := F9J1108 v2 (AC1750 DB Wi-Fi)
+  IMAGE_SIZE := 14464k
+  DEVICE_PACKAGES += kmod-ath10k-ct ath10k-firmware-qca988x-ct kmod-usb2 \
+	kmod-usb3 kmod-usb-ledtrig-usbport
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x50000
+  COMPILE := loader-$(1).bin loader-$(1).uImage
+  COMPILE/loader-$(1).bin := loader-okli-compile
+  COMPILE/loader-$(1).uImage := append-loader-okli $(1) | pad-to 64k | lzma | \
+	uImage lzma
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49
+  IMAGES += factory.bin
+  IMAGE/factory.bin := append-kernel | pad-to $$$$(BLOCKSIZE) | append-rootfs | \
+	pad-rootfs | check-size | edimax-headers F9J1108v1 BR-6679BAC | pad-to $$$$(BLOCKSIZE)
```

Comment:
You really need to pad _after_ the headers?

Reply:
Strangely, yes.  For some reason the checksum calculation is off when the data size is exactly 5242880 bytes (500000 hex), I had to add the padding so that the DCRC error went away.